src/libstd/macros.rs

```diff
@@ -33,6 +33,8 @@
 /// panic!(4); // panic with the value of 4 to be collected elsewhere
 /// panic!("this is a {} {message}", "fancy", message = "message");
 /// ```
+///
+/// [`Box<Any>`]: ../../collections/boxed/struct.Box.html
```

Comment:
I suggest you look into [linkchecker](https://github.com/rust-lang/rust/tree/master/src/tools/linkchecker).

In my opinion the `std` docs are most important so as long as the links in that stay within `std` it shouldn't matter too much about the other crates.